modules/tinymce/src/plugins/paste/main/ts/core/Clipboard.ts

```diff
@@ -127,7 +127,7 @@ const isValidDataUriImage = (editor: Editor, imgElm: HTMLImageElement) => {
 };
 
 const extractFilename = (editor: Editor, str: string) => {
-  const m = str.match(/([\s\S]+?)\.(?:jpeg|jpg|png|gif)$/i);
+  const m = str.match(/([\s\S]+?)\.(?:.+)$/i);
```

Comment:
This doesn't quite seem correct, as we don't want to match anything for the file extension part, we only want to handle what would be a valid file extension (eg this would allow spaces in the extension). I'd suggest limiting this to at least `\w+` instead of `.+`.

- [ ] Restrict this regex more to better match valid file extensions

Reply:
As discussed in slack:

```suggestion
  const m = str.match(/([\s\S]+?)\.(?:[a-z0-9.]+)$/i);
```